### PR TITLE
Openclonk

### PR DESCRIPTION
**Pull request by rardiol**
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I believe the following counts as unfree?

> by David Oerther <A9> Copyright 2015
> All rights reserved, note that the soundtrack does not use the creative common or similar licenses.
> 
> The game Open Clonk ( http://www.openclonk.org/ ) may use and distribute the soundtrack, as long as credit is given. This explicitly includes packaged distributions of the
> game. In-game videos of Open Clonk and game modifications and extensions based on the Open Clonk file-types (.ocd, .ocs and .ocf) don't require credits.
> 
> For all other usage please contact: david.oerther@directbox.com

### CODE REVIEW

**Comment by RamKromberg** [comment]
It's no less free then the GPL telling you the same

File: pkgs/games/openclonk/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+in stdenv.mkDerivation rec {
+  version = "7.0";
+  name = "openclonk-${version}";
+
+  src = fetchurl {
+    url = "http://www.openclonk.org/builds/release/7.0/openclonk-${version}-src.tar.bz2";
+    sha256 = "0ch71dqaaalg744pc1gvg6sj2yp2kgvy2m4yh6l7ljkpf8fj66mw";
+  };
+
+  postInstall = ["mv -v $out/games/openclonk $out/bin/ ;"] ++
```
Comment (joachifm) [inline pkgs/games/openclonk/default.nix]: Equivalently,
```nix
let # for syntax highligthing only
postInstall = ''
  mv ...
'' + optionalString enableSoundtrack ''
  cp ...
'';

File: pkgs/games/openclonk/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  buildInputs = [
+    cmake gnome3.gtksourceview pcre freetype glew gtk3 libjpeg libpng SDL
+    SDL_mixer libupnp tinyxml xorg.libpthreadstubs libxkbcommon xorg.libXdmcp
+    pkgconfig gtest gmock readline epoxy at_spi2_core dbus libxml2
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Free multiplayer action game in which you control clonks, small but witty and nimble humanoid beings";
+    homepage = "http://openclonk.org";
+    license = [ licenses.free ] ++ optional enableSoundtrack licenses.unfree;
```
Comment (joachifm) [inline pkgs/games/openclonk/default.nix]: Makes more sense as `if foo then unfree else free`?

**Comment by joachifm** [comment]
Why are two versions of this required?

**Comment by joachifm** [comment]
The soundtrack license text doesn't say anything about modified versions, so `free` may be misleading consumers who associate "free" with libre software princples. To be safe, we should use at least `unfreeRedistributable` for the soundtrack. The wording "the game may use and distribute the soundtrack" suggests to me that the soundtrack cannot be distributed independently of the game, but that may just be unfortunate phrasing or poor reading comprehension on my part.

If users were able to download the soundtrack at their own discretion, we'd be able to sidestep the whole issue.

File: pkgs/games/openclonk/default.nix (hunk 2 above)
Comment (joachifm) [inline pkgs/games/openclonk/default.nix]: The software appears to be licensed under the ISC license.

**Comment by RamKromberg** [comment]
> The wording [,,,]suggests to me that the soundtrack cannot be distributed independently

I think that's intentional. But the license tag should still be limited to software only and should be considered free here.

A couple of examples of what you're getting into if you overreach into assets:

1. Man pages are often copyrighted and limited to the man pages alone. Quite often, you can't print them and sell them as a book for instance.

2. CC often comes with non-commercial attribution similarly limiting the use case. More importantly CCv3 added the broadcast\rebroadcast clause since the laws in the majority of countries require specific allotments for broadcasting. That's to say, CCv2 has an implicit medium restriction much like this soundtrack limits itself to the game alone which means it was unfree for television.

3. Open source free hardware releases schematics (CAD files) and allow you to modify and reprint. However, the parts they embed are often closed components (micro controllers and SoCs). More so, the tools (like the PCB printer and it's property drivers) that produce the board are almost always closed source. In the software world you write a shim but in the hardware world there's no equivalent so there's no concept libre hardware the way libre software works.

Point is, there might be room for a second assetsLicense clause. But using the current license meta tag usage to include assets will be a real headache.

**Comment by rardiol** [comment]
There are 2 versions because it's a multiplayer game and i wanted to play with someone with 5.4 installed, so after I already wrote the derivations for both 7 and 5.4 I might as well commit them.

**Comment by rardiol** [comment]
Edited accordingly to joachifm's suggestions

File: pkgs/games/openclonk/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    cmake gnome3.gtksourceview pcre freetype glew gtk3 libjpeg libpng SDL
+    SDL_mixer libupnp tinyxml xorg.libpthreadstubs libxkbcommon xorg.libXdmcp
+    pkgconfig gtest gmock readline epoxy at_spi2_core dbus libxml2
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Free multiplayer action game in which you control clonks, small but witty and nimble humanoid beings";
+    homepage = "http://openclonk.org";
+    license = if enableSoundtrack then licenses.unfreeRedistributable else licenses.isc;
+    platforms = platforms.all;
```
Comment (Mic92) [inline pkgs/games/openclonk/default.nix]: do you want to maintain this?

Comment (rardiol) [inline pkgs/games/openclonk/default.nix]: No.

**Pull request closed without merging**